### PR TITLE
Kullbck leibler divergence

### PR DESCRIPTION
**Pull request by trishorts**
This similarity comparison has be used by Nesvizhskii to compare isotope distributions in match between runs FDR determination. 

https://pubmed.ncbi.nlm.nih.gov/33813065/

I added it the spectrum similarity library because it already had the intensity pairs and the normalization functionality that was needed.

The equation and unit test I got from wikipedia:

https://en.wikipedia.org/wiki/Kullback%E2%80%93Leibler_divergence

### CODE REVIEW

File: mzLib/Test/TestSpectralSimilarity.cs [hunk 1]
```diff
@@ -353,5 +353,19 @@ public void TestAllSpectrumSimilaritiesWithDefaultedMzFilter()
             Assert.That(s.PearsonsCorrelation(), Is.Null);
             Assert.That(s.DotProduct(), Is.Null);
         }
+
+        [Test]
+        public void TestKullbackLeiblerDivergence()
```
[the remark this answers is not in the document]
Comment (avcarr2) [reply mzLib/Test/TestSpectralSimilarity.cs:358]: Probably just throw a new ArgumentException if Item2 < 0.

[the remark this answers is not in the document]
Comment (trishorts) [reply mzLib/Test/TestSpectralSimilarity.cs:358]: thoughts on what I should do is someone passed negative mz or intensity?

**Review by avcarr2 — APPROVED** [review]

File: mzLib/Test/TestSpectralSimilarity.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (trishorts) [reply mzLib/Test/TestSpectralSimilarity.cs:358]: blocked these all earlier in the process and added unit tests.

**Review by Yuling1996 — APPROVED** [review]

**Review by Alexander-Sol — APPROVED** [review]
In the paper linked, they use a Poisson distribution for Q. Also, they log10 transform the divergence. Things to think about when using the KL divergence. 

Code looks solid

**Pull request merged**